tensorflow/core/kernels/segment_reduction_ops_impl.h

```diff
@@ -410,15 +414,29 @@ struct UnsortedSegmentFunctor<CPUDevice, T, Index, InitialValueF, ReductionF> {
         }
       }
     };
-
+    auto reductionWorker1D = [&](int64_t begin, int64_t end) -> void {
+      for (int64_t i = 0; i < N; i++) {
+        Index j = internal::SubtleMustCopy(segment_ids(i));
+        // If `j` is in work scope of this worker, do the reduction.
+        if (j >= begin && j < end) {
+            reduction(data_ptr[i], out_ptr[j]);
+        }
+      }
+    };
     // Reduction functors includes Sum, Max, Min, etc. Simply consider it
     // will cost 5 cycles per operation.
     const int64_t kAverTaskSize = num_real_segment / num_segments;
     const int64_t compute_cycles = 5 * inner_dim * kAverTaskSize;
     const int64_t input_bytes = sizeof(T) * inner_dim * kAverTaskSize;
     const int64_t output_bytes = sizeof(T) * inner_dim * kAverTaskSize;
     const Eigen::TensorOpCost cost(input_bytes, output_bytes, compute_cycles);
-    cpu_device.parallelFor(num_segments, cost, reductionWorker);
+    if(is_inner_dim_1d) {
+        cpu_device.parallelFor(num_segments, cost, reductionWorker1D);
+    }
+    else {
+        cpu_device.parallelFor(num_segments, cost, reductionWorker);
+    }
```

Comment:
Nit: Please apply code formatter. The formatted code will probably look like:
```suggestion
    if (is_inner_dim_1d) {
      cpu_device.parallelFor(num_segments, cost, reductionWorker1D);
    } else {
      cpu_device.parallelFor(num_segments, cost, reductionWorker);
    }
```

Alternatively, we could also do
```
    cpu_device.parallelFor(
        num_segments, cost,
        is_inner_dim_1d ? reductionWorker1D : reductionWorker);
```

Reply:
The types are different, so you can't combine them in a ternary expression.